Mathlib/RingTheory/UniqueFactorizationDomain/NormalizedFactors.lean

```diff
@@ -297,6 +310,25 @@ theorem normalizedFactors_multiset_prod (s : Multiset α) (hs : 0 ∉ s) :
     · apply Multiset.prod_ne_zero
       exact fun h ↦ hs (Multiset.mem_cons_of_mem h)
 
+variable {β : Type*} [CancelCommMonoidWithZero β] [NormalizationMonoid β]
+  [UniqueFactorizationMonoid β] {F : Type*} [EquivLike F α β] [MulEquivClass F α β] {f : F}
+
+/--
+If the monoid equiv `f : α ≃* β` commutes with `normalize` then, for `a : α`, it yields a
+bijection between the `normalizedFactors` of `a` and of `f a`.
+-/
+def normalizedFactorsEquiv [DecidableEq α] (he : ∀ x, normalize (f x) = f (normalize x)) (a : α) :
+    {x | x ∈ normalizedFactors a} ≃ {y | y ∈ normalizedFactors (f a)} :=
+  Equiv.subtypeEquiv f fun x ↦
+    if ha : a = 0 then by simp [ha] else by
+      simp [mem_normalizedFactors_iff' ha,
+        mem_normalizedFactors_iff' (EmbeddingLike.map_ne_zero_iff.mpr ha), map_dvd_iff_dvd_symm,
+        MulEquiv.irreducible_iff, he]
+
+theorem normalizedFactorsEquiv_apply [DecidableEq α] (he : ∀ x, normalize (f x) = f (normalize x))
```

Comment:
This can be `simp`, and we probably want `sym_apply` too.

Reply:
I added the `symm` version and `simp` tags. Note however that I had to mark both as `nolint simpNF` since there is some (non obvious to me) simplification somehow that can be done to the LHS

Reply:
What's happening is that `(normalizedFactorsEquiv he a) ⟨p, hp⟩` in the statement has type `↑{y | y ∈ normalizedFactors (f a)}`, where the arrow is the coercion from `Finset` to `Type`. `simp` changes this to the corresponding type `{ x // x ∈ normalizedFactors (f a) }`. Thinks is a good idea to change `normalizedFactorsEquiv` to
```lean
def normalizedFactorsEquiv [DecidableEq α] (he : ∀ x, normalize (f x) = f (normalize x)) (a : α) :
    {x // x ∈ normalizedFactors a} ≃ {y // y ∈ normalizedFactors (f a)} :=
```
and then `nolint simpNF` is not needed anymore.

Reply:
Oh, indeed. That makes sense! Thanks for the review